elements/pf-select/pf-select-option.ts

```diff
@@ -0,0 +1,199 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import type { PropertyValues } from 'lit';
+import { queryAssignedNodes } from 'lit/decorators/query-assigned-nodes.js';
+import { property } from 'lit/decorators/property.js';
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
+
+import styles from './pf-select-option.css';
+
+/**
+ * Option within a listbox
+ * @slot -
+ *        option text
+ * @slot icon
+ *        optional icon
+ * @slot description
+ *        optional description
+ */
+@customElement('pf-select-option')
+export class PfSelectOption extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * whether option is disabled
+   */
+  @property({ reflect: true, attribute: 'aria-disabled', type: String }) ariaDisabled = 'false';
+
+  /**
+   * whether list items are arranged vertically or horizontally;
+   * limits arrow keys based on orientation
+   */
+  @property({ attribute: false, reflect: true }) value: unknown;
+
+  /**
+   * whether option is selected
+   */
+  @property({ attribute: 'selected', type: Boolean }) selected = false;
+
+  /**
+  * total number of options
+  */
+  @property({ type: Number }) setSize!: number;
```

Comment:
```suggestion
  @property({ type: Number }) setSize = 0;
```
naming? what does the useer care that we have a `Set` underneath? and shouldn't this be a getter?